packages/broker/src/plugins/operator/MaintainTopologyHelper.ts

```diff
@@ -3,108 +3,107 @@ import { Provider } from "@ethersproject/providers"
 import { operatorABI, sponsorshipABI } from "@streamr/network-contracts"
 import type { Operator, Sponsorship } from "@streamr/network-contracts"
 import { EventEmitter } from "eventemitter3"
-import { Logger, TheGraphClient } from "@streamr/utils"
+import { EthereumAddress, Logger, TheGraphClient, toEthereumAddress } from "@streamr/utils"
 import { OperatorServiceConfig } from "./OperatorPlugin"
+import { StreamID, toStreamID } from '@streamr/protocol'
+import fetch from 'node-fetch'
 
-/**
- * Events emitted by {@link MaintainTopologyHelper}.
- */
-export interface MaintainTopologyHelperEvents {
-    /**
-     * Emitted if an error occurred in the subscription.
-     */
-    error: (err: Error) => void
+const logger = new Logger(module)
 
+function toStreamIDSafe(input: string): StreamID | undefined {
+    try {
+        return toStreamID(input)
+    } catch {
+        return undefined
+    }
+}
+
+export interface MaintainTopologyHelperEvents {
     /**
      * Emitted when staking into a Sponsorship on a stream that we haven't staked on before (in another Sponsorship)
      */
-    addStakedStream: (streamIds: string[]) => void
+    addStakedStream: (streamIds: StreamID[]) => void
 
     /**
-     * Emitted when a unstaked from ALL Sponsorships for the given stream
+     * Emitted when un-staked from all Sponsorships for the given stream
      */
-    removeStakedStream: (streamId: string) => void
+    removeStakedStream: (streamId: StreamID) => void
 }
 
 export class MaintainTopologyHelper extends EventEmitter<MaintainTopologyHelperEvents> {
```

Comment:
I'd consider putting `eventEmitter` as a field instead of extending `EventEmitter`, as event handling is not a core part of this helper.

Reply:
Future improvement imo